### PR TITLE
Oc 7855 Create a mapping between layouts & items

### PR DESCRIPTION
**Pull request by jkeremian**
Create a mapping between layouts & items
Also , update FormOids for AuditLog and DiscrepancyNote Entities from crf version Oid to crf Oid

### CODE REVIEW

File: core/src/main/java/org/akaza/openclinica/bean/extract/odm/ClinicalDataReportBean.java [hunk 1]
```diff
@@ -136,9 +136,10 @@ public void addNodeClinicalData(boolean header, boolean footer) {
                 for (ExportFormDataBean form : forms) {
                     xml.append(indent + indent + indent + indent + "<FormData FormOID=\"" + StringEscapeUtils.escapeXml(form.getFormOID()));
                     if ("oc1.2".equalsIgnoreCase(ODMVersion) || "oc1.3".equalsIgnoreCase(ODMVersion)) {
-                        String crfVersion = form.getCrfVersion();
-                        if (crfVersion != null && crfVersion.length() > 0) {
-                            xml.append("\" OpenClinica:FormLayoutOID=\"" + StringEscapeUtils.escapeXml(crfVersion));
+                        String formLayout = form.getFormLayout();
+
+                        if (formLayout != null && formLayout.length() > 0) {
```
Comment (yogi-oc) [inline core/src/main/java/org/akaza/openclinica/bean/extract/odm/ClinicalDataReportBean.java]: use Apache StringUtils.isNotEmpty

File: core/src/main/java/org/akaza/openclinica/bean/managestudy/EventDefinitionCRFBean.java [hunk 2]
```diff
@@ -579,38 +577,38 @@ public void setEventName(String eventName) {
...
+
+    public void setDefaultCRF(FormLayoutBean defaultCRF) {
+        this.defaultCRF = defaultCRF;
+    }
 
-	public void setDefaultCRF(CRFVersionBean defaultCRF) {
-		this.defaultCRF = defaultCRF;
-	}
+    public boolean isParticipantForm() {
+        return participantForm == true;
```
Comment (yogi-oc) [inline core/src/main/java/org/akaza/openclinica/bean/managestudy/EventDefinitionCRFBean.java]: Not sure if I understand. Why do we need 
 participantForm == true;?

Can't you just return  participantForm?

File: core/src/main/java/org/akaza/openclinica/bean/managestudy/EventDefinitionCRFBean.java [hunk 3]
```diff
@@ -579,38 +577,38 @@ public void setEventName(String eventName) {
...
-	}
+    public void setParticipantForm(boolean participantForm) {
+        this.participantForm = participantForm;
+    }
 
-	public void setParticipantForm(boolean participantForm) {
-		this.participantForm = participantForm;
-	}
+    public boolean isAllowAnonymousSubmission() {
+        return allowAnonymousSubmission == true;
```
Comment (yogi-oc) [inline core/src/main/java/org/akaza/openclinica/bean/managestudy/EventDefinitionCRFBean.java]: same as above

File: core/src/main/java/org/akaza/openclinica/bean/managestudy/EventDefinitionCRFBean.java (hunk 2 above)
Comment (jkeremian) [inline core/src/main/java/org/akaza/openclinica/bean/managestudy/EventDefinitionCRFBean.java]: This was done by Ron back in 2015 . This is his pull request https://github.com/OpenClinica/OpenClinica/commit/6e5bff933200e0e9c8d00c84eb46f97e10f9983f
Don't know what was the rational behind it.

File: core/src/main/java/org/akaza/openclinica/bean/extract/odm/ClinicalDataReportBean.java (hunk 1 above)
Comment (jkeremian) [inline core/src/main/java/org/akaza/openclinica/bean/extract/odm/ClinicalDataReportBean.java]: I will update it.

File: core/src/main/java/org/akaza/openclinica/bean/managestudy/EventDefinitionCRFBean.java (hunk 3 above)
Comment (jkeremian) [inline core/src/main/java/org/akaza/openclinica/bean/managestudy/EventDefinitionCRFBean.java]: same reply as above.

**Pull request merged**